### ISSUE TITLE
Wrong calculation for sleep-per-month: September 31st missing?

### ISSUE THREAD

**Issue by IzzySoft** [Freeyourgadget/Gadgetbridge#1665]
#### Before reporting a bug, please confirm the following:
- [x] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [x] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

### I got Gadgetbridge from:
* [x] F-Droid
* [ ] I built it myself from source code (specify tag / commit)

#### Your issue is:
I just found out that I didn't sleep at all on September 31st. Well, it's rather Gadgetbridge found out: the value for lack-of-sleep increased by more than 8h from September 30th to October 1st. And the only logical explanation is: An entire day without sleep. But there wasn't, as all days show sleep data, and quite correctly at that.

Could it be that monthly sleep data is somehow based on the assumption that a month has 31 days – instead of counting the columns shown? I've cross-checked that for June (also has 30 days), and it was not confirmed (data matched reality there). So is this only assumed for September?

Cross-checking for missing data: Nope. When switching to weekly, it seems to fit. Or rather it says I overslept 🤣

> **TL;DR:** Monthly calculation looks like based on the assumption there were a September 31st without any sleep.

This is just a minor nuisance – but if someone with the proper knowledge could check with the code, it would be much appreciated. I'm filling my device data below, but I don't think this is device specific.

#### Your wearable device is:
Pebble Time Steel (last available official FW)

#### Your android version is:
7.1.2

#### Your Gadgetbridge version is:
0.32.2

**Comment by vanous** [comment]
Hmm, the weekly/monthly calculations are not really weekly/monthly but days based: 30 and 7 and there aren't that many changes between 7/30, because it all calculates based on these figures, not really per "month". Sure, there could be a bug somewhere... i will check tonight on another band here with someone, who has better sleep pattern then me :)

**Comment by vanous** [comment]
@IzzySoft , I have checked it day by day here on two devices @30 days (also @7 days)... no issue found - no sleep missing, besides the real one... it is rather strange... :thinking:

**Comment by IzzySoft** [comment]
Argh. Found the bug. In front of the keyboard:

The entire frame is shifted by one day – so far so obvious. But the day shifted out to the left had a pretty good night's sleep, catching up what nights before missed: I'd slept about 14h that day (while during the week, it's rather something around 6h).

So of course with 14h dropping out to the left, and just 6h coming in to the right, that makes a diff of … one night's sleep :man_facepalming:
